src/custom-code-helpers/domain-lock/templates/DomainLockTemplate.ts

```diff
@@ -96,14 +105,13 @@ export function DomainLockTemplate (): string {
                     }
                 }
             }
-               
+
             if (!ok) {
-                data;
-            } else {
-                return;
+                const regExp2 = new RegExp("[{domainDestDiff}]", "g");
+                const domainDest = "{domainDest}".replace(regExp2, "");
+
+                that[document][location] = domainDest;
```

Comment:
I think It would be great to add timeout with infinite loop after changing location.

Current behavior:
- "Hacker" removes `document.location` row
- Now domainLock feature not working

My idea:
- "Hacker" removes `document.location` row
- After 15 seconds (it's enough for redirect, i think) packe just freezes.

```
const stuck = () => {
  for(let i=0; i < 100; i--){
    // some useless code here, maybe the old one:
    // https://github.com/javascript-obfuscator/javascript-obfuscator/blob/32e1dad4bc6a17469182463ec5bc24fe0f31c07d/src/custom-code-helpers/domain-lock/templates/DomainLockTemplate.ts#L10
  }
};

// ...

if(!ok){
  document.location = "...";
}
// ...
if(!ok){
  setTimeout(stuck, 15000);
}
```

But it would be good to make it more obscure, for example, put `setTimeout` separase, after some useless code.

Reply:
I don't mind doing this, but to be honest I think if you're dedicated enough to find the `document.location`, which is already almost impossible, you're going to be able to remove the useless code or `setTimeout` or the whole domain check as well. In which case I think this doesn't really make it better and maybe only makes it easier to find as there is more code you could try to recognize.